client/go/outline/config.go

```diff
@@ -52,6 +52,9 @@ type configJSON struct {
 	Password string `json:"password"`
 	Method   string `json:"method"`
 	Prefix   string `json:"prefix"`
+
+	DynamicKeyServer     string `json:"server"`
```

Comment:
Having many ways to express the same thing in the code is a pain. Please don't.
That's convenient for providers, but makes it hard to write correct code, as evidenced by the bug you are trying to fix. It's also a pain for things like getting the host or address from the config.

We need to normalize the config as soon as we get it, so all our code works with a single single way to represent the config.

I'm in the process of normalizing the config already: https://github.com/Jigsaw-Code/outline-apps/pull/2203

Please revert this change to the Go code and fix in the TS code. BTW, what changed that broke this? I couldn't pinpoint in my PR.

Reply:
Here is the code that broke the behavior: https://github.com/Jigsaw-Code/outline-apps/pull/2192/files#r1750971442

But should TS code care about the format of the config? I think TS will only treat it as an opaque string, and pass this string to Go. So in the end, Go will be the single source of truth that understands the config format.

Reply:
Unfortunately TS needs to know about the config for a few reasons:
- to validate on adding keys
- to extract the server name for display
- to resolve the server name on Electron 
- to access tunnel configuration in the application-specific code

We won't be getting rid of those immediately. We may move some of that to Go, but it will require a new Go target. For now, let's stick to TS, and pass the normalized config to Go.

For the issue you are trying to fix, let's fix where we parse the JSON in Typescript.